### PR TITLE
Further Travis improvements

### PR DESCRIPTION
**Pull request by neongreen**

### CODE REVIEW

**Comment by neongreen** [comment]
This gets us to 25 minutes, but we should be able to get 20.

File: .travis.yml [hunk 1]
```diff
@@ -4,15 +4,8 @@ dist: trusty
 language: generic
 
 addons:
-  apt:
```
Comment (neongreen) [inline .travis.yml:7]: Moved to individual stages.

File: .travis.yml [hunk 2]
```diff
@@ -45,37 +40,46 @@ jobs:
 
     - stage: "Build the backend and upload a Docker image"
       before_script:
+        - sudo apt-get install -y libgmp-dev
         - curl -sSL https://get.haskellstack.org/ | sh
+        - stack setup
```
Comment (neongreen) [inline .travis.yml:45]: Installs GHC (same as before but it's nicer to do it as a separate step and get Travis timings for this step)

File: .travis.yml [hunk 3]
```diff
@@ -45,37 +40,46 @@ jobs:
...
+        # Upload the Docker image
         - |
           if [ "$TRAVIS_EVENT_TYPE" = "push" ]; then
             export BRANCH="$(echo "$TRAVIS_BRANCH" | tr '/' '-')"
             make back/travis-docker "tag=$BRANCH--back"
             docker login quay.io -u "$DOCKER_USER" -p "$DOCKER_PASS"
             docker push "quay.io/aelve/guide:$BRANCH--back"
           fi
-
-    # Our goal is full documented code with ZERO undocumented definitions.
```
Comment (neongreen) [inline .travis.yml:60]: I consolidated this step and the previous step.

**Comment by neongreen** [comment]
This PR brings us down to 23–25 minutes.

File: .travis.yml [hunk 4]
```diff
@@ -4,15 +4,8 @@ dist: trusty
...
 
 addons:
-  apt:
-    packages:
-    # GHC
-    - libgmp-dev
-    # Selenium
-    - xvfb
-    # Testcafe
-    - fluxbox
```
Comment (willbasky) [inline .travis.yml:14]: Don't we need these packages anymore?

Comment (neongreen) [reply .travis.yml:14]: We do, but now we install them only when we need them:

image

**Review by willbasky — APPROVED** [review]

**Pull request merged**